proto/valve.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+
+package smartcore.bos;
+
+option go_package = "github.com/vanti-dev/sc-bos/pkg/gen";
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+
+service ValveApi {
+  rpc GetValve(GetValveRequest) returns (Valve) {}
+  rpc PullValve(PullValveRequest) returns (stream PullValveResponse) {}
+  rpc UpdateValve(UpdateValveRequest) returns (Valve) {}
+}
+
+message Valve {
```

Comment:
A valve is a type of device, I'm not convinced it describes a trait. Maybe `Flow` is more apt as a trait. A valve controls the _flow_ of something. Flow has a rate and direction, it can have pressure, it can be open/closed, etc.